src/sentry/utils/kvstore/bigtable.py

```diff
@@ -235,11 +236,24 @@ def _set(self, key: str, value: bytes, ttl: timedelta | None = None) -> None:
         # tracking now is whether compression is on or not for the data column.
         flags = self.Flags(0)
 
+        metrics.distribution(
+            "storage.put.size",
+            len(value),
+            tags={"usecase": "nodestore", "compression": "none"},
+            unit="byte",
+        )
         if self.compression:
             compression_flag, strategy = self.compression_strategies[self.compression]
             flags |= compression_flag
             value = strategy.encode(value)
 
+            metrics.distribution(
+                "storage.put.size",
+                len(value),
+                tags={"usecase": "nodestore", "compression": self.compression},
```

Comment:
I’m aware this might not be the right place for this metric, as this is the more generic bigtable KV abstraction.
However this is the only place we have access to un/compressed sizes, and AFAIK, nodestore is the only user of bigtable right now.